### DISCUSSION TITLE
[FEAT] A toggle in admin panel to not let user change the url

### DISCUSSION THREAD

**Discussion by abhay-ramesh** [JulianPrieber/littlelink-custom#491] [Feature requests]

**Comment by JulianPrieber** [comment]
Huh?

**Reply by JulianPrieber** [reply]
What URL?

**Reply by brasofilo** [reply]
I supose OP is referring to "Appearance" -> "Page URL", that is the same that I'm wondering if we can block regular users from modifying:
![image](https://github.com/LinkStackOrg/LinkStack/assets/1542052/ca27e137-2d3b-480f-904a-cd79bae742e7)

**Reply by JulianPrieber** [reply]
I suppose that makes sense?

Well, that's probably never coming to mainline. Maybe a fork?
If someone wants to do that, that's like commenting one line. Maybe two if you want to make it secure.

**Reply by JulianPrieber** [reply]
You very easily get into situations with this where users don't have a page URL set.

**Reply by brasofilo** [reply]
My use case is where I'm setting up the user's accounts/URLs and I don't want them messing with it.

I modified the `<input>` line like this if anyone is interested in a brute force fix.
File `/resources/views/studio/page.blade.php`, line 226:

```
<?php $readonly = isset($_GET['edit']) ? '' : 'readonly'; ?>
@if(auth()->user()->role == 'admin' || auth()->user()->role == 'vip')
  <?php $readonly = ''; ?>
@endif
<input type="text" class="form-control" name="littlelink_name" value="{{ $page->littlelink_name ?? '' }}" required <?php echo $readonly; ?>>
```
Admins and VIPs can edit the field.
Adding "?edit" to the URL allows editing the field: `https://links.mylinkstack.com/studio/page?edit`
